\documentclass[letterpaper]{article}

\usepackage[utf8]{inputenc}
\usepackage[sort, comma]{natbib}
\usepackage{alifexi}
\usepackage[bottom]{footmisc}
\usepackage[colorlinks=true,citecolor=green,linkcolor=blue]{hyperref}

\usepackage{mathtools}
\usepackage{commath}

\usepackage{dblfloatfix}

\title{Reproducing and updating results from
\textit{Uncovering disease-disease relationships\\through the incomplete interactome}}
\author{Robin Petit$^{1}$ \and Tom Lenaerts$^{1}$\\
\mbox{}\\
$^1$Universit√© Libre de Bruxelles}


\begin{document}
\maketitle

\begin{abstract}
  This paper intends to work on the results exposed in the paper \textit{Uncovering
	disease-disease relationships through the incomplete interactome}~\citep{originalPaper}
	in order to test their reproducibility, and update the different datasets to check if the
	presented procedure, which intends to be a systematic analysis, still stands for a more
	recent version of the interactome, and if the results exposed have shifted towards a more
	or less significant level. We find that
\end{abstract}

\section{Introduction}
An interactome is a graph containing all the molecular interactions within a cell. This notion appeared
in 1999 (for the drosophila)~\citep{sanchez1999grasping} and the need to thoroughly study this structure has
been expressed less than 15 years ago~\citep{UnderstandingTheCellFunctionalOrganization}.

The interactome presented in the original paper~\citep{originalPaper} has been used because authors showed that it is
sufficiently complete to be systematically studied. Even though the main idea was to study the diseases
behaviour inside the graph, different studies can be applied, e.g. drug-disease correlation~\citep{Yu2016extraction},
or digenic diseases~\citep{gazzo2015dida}.

In the original paper, authors applied disease genes databases (in particular OMIM~\citep{amberger2008OMIM}
and GWAS compiled by PhenGenI~\cite{ramos2014PhenGenI}) on the human interactome in order to determine
the properties of their distribution in the graph. Major results were that: firstly diseases tend to
\textit{cluster} in denser subgraphs than the interactome itself (shown by bigger largest connected
component than expected in random interactome subgraphs), secondly that phenotypically close diseases
tend to overlap on a significant amount of genes.

Authors of the original paper also remind that the interactome is incomplete and estimated around $20\%$-complete
for the interactions by the estimation that the interactome contains roughly $6.5 \times 10^5$ interactions,
and around $54\%$-complete for the proteins involved by the estimations that the interactome contains roughly
$2.5 \times 10^4$ nodes~\citep{ATruerMeasureOfOurIgnorance,estimatingTheSizeOfTheHumanInteractome}

\underline{NOTE:} references expressed as Sx refer to the original paper's supplementary materials. Any
other reference is to this very paper, unless explicitly mentioned.

\section{Reproducing results}
The first part of this paper focuses on the reproduction of exposed results in the original paper,
namely the disease modules propensity to cluster into highly connected components and the significantly
lower separation indicator values for highly gene related disease pairs.

The interactome used in the original paper contains 13460 genes and 141296 physical links constructed on several
interactions databases including BioGRID~\citep{chatr2017biogrid}, IntAct~\citep{kerrien2011intact},
TRANSFAC~\citep{matys2003transfac}, MINT~\citep{licata2011mint}, HPRD~\citep{keshava2008HPRD}, KEGG and
BIGG~\cite{lee2008KEGG-BIGG}, CORUM~\citep{ruepp2009corum} and PhosphitePlus~\citep{hornbeck2011phosphositeplus}
and is available to download as supplementary material.

Diseases from OMIM (Online Mandelian Inheritance in Man) and GWAS (Genome-Wide Asociation Studies)
related to at least 20 genes have been compiled to be applied to the interactome.

	\subsection{Clustering of disease modules}\label{subsec:clustering of disease modules}
	Figure S4.b plots the relative size of each disease module versus its relative size (defined as the
	quotient of the largest connected component size by the number of genes related to the disease).

	When plotting the same data making $10^5$ random simulations per disease and setting the significance
	threshold to be $1.6$\footnote{Considering the distribution to be normal~\citep{fluctuationGiantComponent},
	a $z$-score $\geq 1.6$ represents a $p$-value $\leq 0.05$ which corresponds to considered
	\textit{significant} results.}, the obtained result is shown on
	Figure~\ref{fig:zscore}, which fits the one presented in the original paper.

	\begin{figure}[!h]
		\includegraphics[width=.5\textwidth]{images/S4.b100000.eps}
		\caption{z-score of largest connected component size vs relative module size.\label{fig:zscore}}
	\end{figure}

	\begin{figure*}[!t]
		\hspace{-1.8cm}
		\vspace{-1cm}
		\includegraphics[scale=.35]{images/s_AB_histograms.eps}
		\caption{Disease pairs separation.\label{fig:s_AB histogram}}
	\end{figure*}

	This figure shows that several diseases do not present a significantly larger largest connected component
	than expected by chance, but that these diseases have a thin relative size ($< 0.2$, so that less than $20\%$
	of their related genes are not present in the current interactome). Also, diseases with bigger relative size
	have a higher $z$-score, leading to think that a more complete interactome, with higher coverage of the diseases
	would increase the significance of the result.

	\subsection{Separation distribution}
	Original paper's figure 3.K-L plots the separation distribution of the disease pairs according to their
	overlapping score ($C$-score and $J$-score defined respectively as $\abs {A \cap B}/\min(\abs A, \abs B)$ and
	$\abs {A \cap B}/\abs {A \cup B}$ for $A$ and $B$ two diseases). Disease pairs $AB$ having a null $J$-score
	(and thus $C$-score) do not share any gene since $\abs {A \cap B} = 0$. Disease pairs $AB$ having a $C$-score
	equal to 1 are either identical (if their $J$-score equals 1) or one is a strict subset of the other.

	Figure~\ref{fig:s_AB histogram} plots the distribution of $s_{AB}$ separation indicator.
	Distribution shown on ({\bf A}) and ({\bf B}) fit almost exactly the one presented in the article, the only difference
	being that for non-overlapping disease pairs, the amount of pairs having a separation value below 0 is $710$
	versus $717$ in the original paper, being totally non-significant since $7$ pairs represent less than
	$0.03\%$ of all the non-overlapping pairs set.


\section{Databases update}

	\subsection{Interactome update}
	In order to update the interactome, the tool \textit{inter-tool}~\citep{inter-tools} has been used.
	Latest datasets from BioGRID, IntACT, and the Database of Interacting Proteins (DIP)
	(on July 25th) and of MINT (on July 27th) have been downloaded and merged via inter-tools.

	Inter-build (one of the programs from Inter-tools) requires datasets in the PSI-MITAB format, an extension of the
	PSI-MI format~\citep{MITABFormat}, and outputs a tsv file.

	The outputted file by Inter-build and the interactome provided with the original paper both use Entrez gene IDs, they
	can therefore easily be merged. In order to merge these, the script \texttt{merger.py} has been written. As these two
	tsv files have a different format (different columns), only the gene IDs are outputted by \texttt{merger.py}.

	This newer version merged with the original interactome yield a new graph having 17786 nodes and 370326 edges
	(so a bit more than 1.3 times the initial amount of nodes, and more than 2.6 times the initial amount of edges).

	This newer version is around $71\%$-complete for the proteins involved and $57\%$-complete for the
	interactions~\citep{estimatingTheSizeOfTheHumanInteractome,ATruerMeasureOfOurIgnorance}.

	\subsection{Disease genes update}
	Disease genes considered in this update are the same 299 diseases studied in the original paper. An update of these
	is yet to be done in further investigations.

	\subsection{Updated results}
	By defining the density of a graph $\Gamma = (V, E)$ as being $d(\Gamma) \coloneqq \abs E/\binom {\abs V}2$,
	we find that the newer interactome is more than 1.5 times denser than the original one ($0.156\%$ versus $0.234\%$
	for the newer one).

	\begin{figure}[!h]
		\includegraphics[width=.5\textwidth]{images/new_interactome_S4.b100000.eps}
		\caption{z-score of the largest connected component size vs relative module size of the newer interactome.
		\label{fig:new interactome zscore}}
	\end{figure}

	Figure~\ref{fig:new interactome zscore} is the adaptation of Figure~\ref{fig:zscore} for the newer interactome.
	We observe that 12 more diseases have a $z$-score below the threshold of $1.6$, which makes results less significant,
	whereas general relative size has shifted towards right as seen in Figure~\ref{fig:rel sizes comparison}.

	\begin{figure}[!h]
		\includegraphics[width=.5\textwidth]{images/rel_sizes_comparison.eps}
		\caption{Comparison of relative size distribution between original and newer interactomes.\label{fig:rel sizes comparison}}
	\end{figure}

	This is explained by the increase in the gene number, leading to a higher coverage of the disease genes. Also, $z$-score
	maximum has dropped from 31.6 to 27.5, the mean $z$-score has increased from 6.2 to 6.4 due in part to the density
	increase in the region $z$-score $\in [10, 20]$.

	As well, average relative size for the given diseases has increased from $22\%$ to $32\%$ in the newer interactome,
	showing the better coverage allowed by the more recent version.

	The interpretation of Subsection~\ref{subsec:clustering of disease modules} still stand: diseases having a low $z$-score still
	have a low relative size, and diseases with a higher $z$-score have a higher relative size. So either the interactome is still
	too incomplete for these diseases, or they lie in very sparse regions of the interactome.

	Due to the density increase, the degree distribution has changed as well. Figure~\ref{fig:degree distribution comparison}
	shows the degree distribution of both the original interactome and the new one. The most visible change is that the newer
	interactome contains more connected nodes which implies a mean twice as big.

	\begin{figure}[!h]
		\includegraphics[width=.5\textwidth]{images/degree_distributions_comparison.eps}
		\caption{Degree distribution comparison.\label{fig:degree distribution comparison}}
	\end{figure}

	\begin{figure*}%[!t]
		\hspace{-1.8cm}
		\vspace{-1cm}
		\includegraphics[scale=.35]{images/new_interactome_s_AB_histogram.eps}
		\caption{Disease pairs separation in the new interactome.\label{fig:new interactome s_AB}}
	\end{figure*}

	Yet, the network is still scale-free, and its degree distribution still follows a power law, which is inherent to
	biological networks (coefficient 1.6 for the newer one versus 1.53 for the original one).\footnote{Determined by
	taking coefficient $\gamma$ in the relation $\log(P(k)) \sim -\gamma\log(k)$ found by a linear regression.}
	$\gamma$ is bigger than 1 and smaller than 3, which is considered
	relevant~\citep{UnderstandingTheCellFunctionalOrganization,InteractomeNetworksAndHumanDisease}.

	The 12 diseases which have a decrease of $z$-score below 1.6 is due to the increase of the interactome density implying
	that a subgraph taken at random tends to have a wider LCC at equal size.

	Separation analysis applied on the newer interactome yields results presented in Figure~\ref{fig:new interactome s_AB}.
	We observe therefore that non-overlapping diseases have a higher separation score in the newer interactome: from 710
	disease pairs to 324 pairs having a $s_{AB} > 0$ score (so $54\%$ of the non-overlapping disease pairs increased their
	separation score above 0).

	We also observe that $6\%$ of the complete subset disease pairs decreased their separation score below 0, which is way
	less significant.

	More generally, separation scores have tightened around 0: $s_{AB}$ is in $[-3.2, 1.6]$ in the original interactome, and
	in the newer one, $s_{AB}$ is in $[-2.5, 1.1]$.

\section{Improvements}

	\subsection{Subgraph largest connected component distribution}
	Figures~\ref{fig:zscore} and~\ref{fig:new interactome zscore} require a null hypothesis in order to define a $z$-score,
	being the random one. Those are computed as follows: if $S_D$ is the disease module associated with a given disease $D$,
	then its $z$-score is given by:
	\begin{equation}
		z\text{-score} = \frac {\abs {S_D} - \mu(S^{\text{rand}})}{\sigma(S^{\text{rand}})},
	\end{equation}
	with $\mu(S^{\text{rand}})$ and $\sigma(S^{\text{rand}})$ being respectively the mean and the
	standard deviation of the largest connected component size of a random subgraph of size $\abs D$
	in the interactome.

	\begin{figure}[!h]\centering
		\includegraphics[width=.45\textwidth]{images/Srand_distribution_1000_sims.eps}
		\caption{$S^{\text{rand}}$ mean and standard deviation distribution.\label{fig:Srand distribution}}
	\end{figure}

	These values are obtained by simulations: taking subgraphs at random of given size in the
	interactome. With $10^3$ simulations per subgraph size, Figure~\ref{fig:Srand distribution}
	plots simulated mean and standard deviations of largest connected component size versus
	subgraph size.

	\subsection{Analytically determined probability density}
	In order to avoid simulation computation time, a probability mass function has been determined. For a graph $\Gamma = (V, E)$
	such that $\abs E = m$ and $\Lambda_k^m(V, \cdot)$, the set of all graphs having $V$ as vertex set, $m$ edges, and a LCC of size $k$,
	we define $p_k$, the probability that $\Gamma$ has LCC of size $k$ as:
	\begin{equation}
		p_k = \abs {\Lambda_k^m(V, \cdot)}/\binom {\binom {\abs V}2}m,
	\end{equation}
	which requires $\abs {\Lambda_k^m(V, \cdot)}$ to be computed. Yet, this set cardinality is defined by a recurrence relation
	(see proof in supplementary materials), which makes computations several orders of magnitude slower (even with dynamic programming
	and caching).

	A Python3 implementation is given in \texttt{source/lcc\_size/}.

\section{Conclusion}

\footnotesize
\bibliographystyle{apalike}
\bibliography{report}{}

\end{document}
